modules/programs/autojump.nix

```diff
@@ -0,0 +1,61 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.programs.autojump;
+in {
+  meta.maintainers = [ maintainers.evanjs ];
+
+  options.programs.autojump = {
+    enable = mkEnableOption "autojump";
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.autojump;
+      defaultText = literalExample "pkgs.autojump";
+      description = "The package to use for the autojump binary.";
+    };
+
+    enableBashIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Bash integration.
+      '';
+    };
+
+    enableZshIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Zsh integration.
+      '';
+    };
+
+    enableFishIntegration = mkOption {
+      default = true;
+      type = types.bool;
+      description = ''
+        Whether to enable Fish integration.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    home.packages = [ cfg.package ];
+
+    programs.bash.initExtra = mkIf cfg.enableBashIntegration ''
```

Comment:
To make it more like the existing Bash setup I would suggest changing this to `mkIf cfg.enableBashIntegration (mkBefore ''`.

Reply:
Ahhh I see it now.
This previously occurred within the `bashrcExtra` block, immediately before `initExtra`.
So using `mkBefore` will again place it just before `initExtra`.

Thanks!